lib/AST/ASTScope.cpp

```diff
@@ -1777,7 +1777,10 @@ DeclContext *ASTScope::getInnermostEnclosingDeclContext() const {
   for (const ASTScope *scope = this; ; scope = scope->getParent()) {
     if (auto dc = scope->getDeclContext()) return dc;
   }
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunreachable-code"
   llvm_unreachable("Top-most scope is a declaration context");
+#pragma clang diagnostic pop
```

Comment:
Ahhh…that makes sense and is also terrible. I wonder if we should sink the pragmas into the `llvm_unreachable` macro. \*thinks out loud, creating work for someone to go talk to LLVM\*